app/assets/javascripts/discourse/app/components/user-menu/menu.js

```diff
@@ -164,6 +183,33 @@ const CORE_BOTTOM_TABS = [
   },
 ];
 
+const CORE_OTHER_NOTIFICATIONS_TAB = class extends UserMenuTab {
+  @tracked groupedUnreadNotifications =
+    this.currentUser.grouped_unread_notifications;
```

Comment:
What this does is it tracks the `groupedUnreadNotifications` property so that if we update like this `this.groupedUnreadNotifications = <blah something>`, Ember will update all properties that depend on it, but it doesn't make the `this.currentUser.grouped_unread_notifications` property tracked which is what we want to track. When Message Bus updates the `grouped_unread_notifications` property of `currentUser`, the `groupedUnreadNotifications` property here will diverge and be stuck with the outdated copy of `grouped_unread_notifications`.